mountpoint-s3/tests/fuse_tests/mod.rs

```diff
@@ -263,3 +287,22 @@ pub fn read_dir_to_entry_names(read_dir_iter: ReadDir) -> Vec<String> {
         })
         .collect::<Vec<_>>()
 }
+
+#[macro_export]
+macro_rules! sleep_till_retry_succeed {
+    ($method_result:expr) => {
+        for i in 1..1000 {
+            match $method_result {
+                Ok(bool_value) => {
+                    if bool_value {
+                        break;
+                    } else {
+                        sleep(Duration::from_millis(10));
+                    }
+                }
+                Err(_) => panic!("The provided method could not provide result"),
+            }
+            assert!(i < 1000, "The provided method could not provide success result");
+        }
+    };
+}
```

Comment:
I think the values and sleeps we have is a bit extreme here.

Each time, we're going to be making a call to S3 (when not using mock client), which will add a bunch of latency to any test if the test fails. In the worst case where the condition never holds, that would be `(1000 * avg_latency) + 10s`.

Maybe we retry five times with a 200ms delay each time? Better yet, we could grow the sleep exponentially, doubling with each retry (up to a small number like five retries).